### PR TITLE
[Docs] Note that certain features are community maintained

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

Adds notes explaining that Ray's support on Azure, Aliyun, and SLURM is community-maintained.
Rephrases the mention of K8s support in the intro.

This PR replaces https://github.com/ray-project/ray/pull/25504. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/cloud.rst [hunk 1]
```diff
@@ -123,7 +126,7 @@ Ray with cloud providers
 
     GCP Node Provider Maintainers (GitHub handles): wuisawesome, DmitriGekhtman, ijrsvt
 
-.. tabbed:: Aliyun
+.. tabbed:: Aliyun (community-maintained)
```
Comment (richardliaw) [inline doc/source/cluster/cloud.rst:129]: Can you instead leave this as a note underneath? This increases the tab-width, and is also inconsistent with Azure.

Comment (DmitriGekhtman) [reply doc/source/cluster/cloud.rst:129]: sgtm -- inconsistency unintentional

Comment (DmitriGekhtman) [reply doc/source/cluster/cloud.rst]: Fixed.

**Review by richardliaw — APPROVED** [review]

**Review by tupui — APPROVED** [review]

**Comment by wuisawesome** [comment]
Should we qualify what non-community supported means? (e.g. we're committed to actively maintaining it, but sunsetting it from a feature development perspective)

**Comment by richardliaw** [comment]
"non-community supported" this isn't a label right?

I don't think we need to introduce any more detail here though.

**Comment by DmitriGekhtman** [comment]
> > . "non-community supported" this isn't a label right?
> 
> yeah it's not a label, but what I mean is should we set expectations for the support we'll provide on AWS and GCP. (e.g. bug fixing, but not really active feature development).

I had to stop and think for a moment, which is good enough rationale for keeping this thought for a future PR :) 
Some reservations:
- We're not thinking about feature work for these stacks any time soon, but we might in theory later.
- Feature-work vs. bug-fixing can sometimes be a bit of a blurry boundary. 
- Mention of "sunset" or "maintenance" can send the wrong message.

**Comment by richardliaw** [comment]
Yeah I thought about this but I couldn't think of a great reason to need to
explicitly set the expectations? If people have features they want to see
developed they can post on github.

Anyways, I'll stop now and am OK with the PR as-is.

**Review by wuisawesome — APPROVED** [review]
discussion result sgtm

**Comment by AmeerHajAli** [comment]
Thanks Alex and Dmitri!

**Pull request merged**